services/github/github-deployments.service.js

```diff
@@ -0,0 +1,122 @@
+'use strict'
+
+const gql = require('graphql-tag')
+const Joi = require('@hapi/joi')
+const { GithubAuthV4Service } = require('./github-auth-service')
+const { documentation, transformErrors } = require('./github-helpers')
+
+const schema = Joi.object({
+  data: Joi.object({
+    repository: Joi.object({
+      deployments: Joi.object({
+        nodes: Joi.array()
+          .items(
+            Joi.object({
+              latestStatus: Joi.object({
+                state: Joi.string(),
+              }),
+            })
+          )
+          .required(),
+      }).required(),
+    }).required(),
+  }).required(),
+}).required()
+
+const greenStates = ['success']
+
+const yellowStates = ['in_progress', 'queued', 'pending']
+
+const redStates = ['error', 'failure', 'unknown']
+
+const blueStates = ['inactive']
+
+module.exports = class GithubDeployments extends GithubAuthV4Service {
+  static get category() {
+    return 'other'
+  }
+
+  static get route() {
+    return {
+      base: 'github/deployments',
+      pattern: ':user/:repo/:environment',
+    }
+  }
+
+  static get examples() {
+    return [
+      {
+        title: 'GitHub deployments',
+        namedParams: {
+          user: 'badges',
+          repo: 'shields',
+          environment: 'production',
```

Comment:
let's using the staging env instead so the example provides a better sample
```suggestion
          environment: 'shields-staging',
```

Reply:
Had not seen this repo also uses deployments :D Sweet, that makes the test-case also a lot easier, tnx :)